tests/base.py

```diff
@@ -223,6 +230,30 @@ def expected_full_table_streams(self):
         return set(stream for stream, rep_meth in self.expected_replication_method().items()
                    if rep_meth == self.FULL)
 
+    @abstractmethod
+    def testable_streams_dynamic(self):
+        pass
+
+    @abstractmethod
+    def testable_streams_static(self):
+        pass
+
+    def testable_streams(self, environment='sandbox', data_type=DataType.DYNAMIC):
+        allowed_environments = {self.SANDBOX, self.PRODUCTION}
+        if environment not in allowed_environments:
+            raise NotImplementedError("Environment can only be one of {}, but {} provided".format(allowed_environments, environment))
+
+        if environment == self.SANDBOX:
+            if data_type == DataType.DYNAMIC:
+                return self.testable_streams_dynamic().intersection(self.sandbox_streams())
+            else:
+                return self.testable_streams_static().intersection(self.sandbox_streams())
+        else:
+            if data_type == DataType.DYNAMIC:
+                return self.testable_streams_dynamic().intersection(self.production_streams())
+            else:
+                return self.testable_streams_static().intersection(self.production_streams())
+
```

Comment:
I think this is a much better solution to dealing with multiple environments and data sets.